internal/handler/authenticate.go

```diff
@@ -100,14 +97,7 @@ func (h *Handler) authenticate(ctx context.Context) error {
 		}
 	}
 
-	if !hasUser {
-		// There is no user in the database, let the backend check the authentication.
-		// We do not want unauthenticated users accessing the database, while allowing
-		// users with valid backend credentials to access the database.
-		// TODO https://github.com/FerretDB/FerretDB/issues/4100
-		conninfo.Get(ctx).UnsetBypassBackendAuth()
-		h.L.Error("backend is used for authentication - no user in admin.system.users collection")
-
+	if !hasUser && conninfo.Get(ctx).LocalPeer() {
```

Comment:
This is the same logic explained in local exception https://www.mongodb.com/docs/manual/core/localhost-exception/#std-label-localhost-exception.

I have a question, do we want to allow local exception _even after the the first user was created_? For integration tests to work, there are these options I can think of:
1) create user in test setup (what's happening currently)
2) use envtool to create user (https://github.com/FerretDB/FerretDB/pull/4133 with lots of SQL)
3) enable local exception even if there is existing user

What's your thoughts?

Reply:
> do we want to allow local exception even after the the first user was created?

No. Why would we want that?

Reply:
That question was trying to figure out how to handle creating users in the test.
We opt for creating user within test setup but making it optional.